frontend/vre/utils/jsonld.model.js

```diff
@@ -1,11 +1,91 @@
 import {APICollection} from "./api.model";
 
 /**
- * Generic subclass of APICollection that parses incoming JSON-LD to a
- * list of subjects.
+ * The @graph property inside JSON-LD
+ * @typedef {Object} JSONLDGraph
+ */
+
+/**
+ * An individual subject definition inside JSON-LD
+ * @typedef {Object} JSONLDSubject
+ */
+
+/**
+ * Generic subclass of APICollection that parses incoming JSON-LD to an
+ * array of all subjects. The contents of subjects are left unchanged.
  */
 export var JsonLdCollection = APICollection.extend({
     parse: function(response) {
+        if (!response.hasOwnProperty("@graph")) {
+            throw "Response has no @graph key";
+        }
         return response["@graph"];
     }
 });
+
+/**
+ * Return a nested version of a given subject by adding to it the objects
+ * it refers to if they are found in the graph.
+ * The subject passed to this function as an argument is not changed.
+ * @param graph{JSONLDGraph} - The full contents of the graph in JSON-LD
+ * @param subject{JSONLDSubject} - The subject including its predicates and objects to create a nested version of
+ * @returns {Object}
+ */
+export function nestSubject(graph, subject) {
+    const recursiveFunction = function(graph, subject, baseSubject) {
+        const transformedSubject = _.clone(subject);
+        for (let property of Object.keys(subject)) {
+            if (subject[property].hasOwnProperty("@id")) {
+                // This is a reference to another subject
+                const refereedSubject = graph.find((thisSubject) => thisSubject["@id"] === subject[property]["@id"]);
+                if (refereedSubject && refereedSubject["@id"] !== baseSubject["@id"]) {
+                    /* If the refereed subject was found in the graph, use it as replacement.
+                       Only do this if the subject is not the same as the one we started with,
+                       to avoid an endless loop. (Alternative would be to create a circular reference) */
+                    transformedSubject[property] = recursiveFunction(graph, refereedSubject, baseSubject);
+                }
+            }
+        }
+        return transformedSubject;
+    }
+    return recursiveFunction(graph, subject, subject);
+}
```

Comment:
Despite all the comments about this function, I should add that I still agree with the general approach. In fact, I like the brevity.